cmake/modules/contrib/CUTLASS.cmake

```diff
@@ -16,16 +16,61 @@
 # under the License.
 
 if(USE_CUDA AND USE_CUTLASS)
-  tvm_file_glob(GLOB CUTLASS_CONTRIB_SRC src/relay/backend/contrib/cutlass/*.cc src/relax/backend/contrib/cutlass/*.cc)
+  set(CUTLASS_GEN_COND "$<AND:$<BOOL:${USE_CUDA}>,$<BOOL:${USE_CUTLASS}>>")
+  set(CUTLASS_RUNTIME_OBJS "")
+
+  tvm_file_glob(GLOB CUTLASS_CONTRIB_SRC
+    src/relay/backend/contrib/cutlass/*.cc
+    src/relax/backend/contrib/cutlass/*.cc
+  )
   list(APPEND COMPILER_SRCS ${CUTLASS_CONTRIB_SRC})
 
   set(FPA_INTB_GEMM_TVM_BINDING ON)
   set(FPA_INTB_GEMM_TVM_HOME ${PROJECT_SOURCE_DIR})
 
-  set(CUTLASS_DIR ${PROJECT_SOURCE_DIR}/3rdparty/cutlass)
+  ### Build cutlass runtime objects for fpA_intB_gemm using its cutlass submodule
   add_subdirectory(${PROJECT_SOURCE_DIR}/3rdparty/cutlass_fpA_intB_gemm)
+  target_include_directories(fpA_intB_gemm PRIVATE
+    ${PROJECT_SOURCE_DIR}/3rdparty/cutlass_fpA_intB_gemm
+    ${PROJECT_SOURCE_DIR}/3rdparty/cutlass_fpA_intB_gemm/cutlass/include
+  )
+  set(CUTLASS_FPA_INTB_RUNTIME_SRCS "")
+  list(APPEND CUTLASS_FPA_INTB_RUNTIME_SRCS src/runtime/contrib/cutlass/moe_gemm.cc)
```

Comment:
Is it needed? It’s already included in the submodule https://github.com/tlc-pack/cutlass_fpA_intB_gemm/blob/main/cutlass_kernels/CMakeLists.txt

Reply:
I see. I originally made my change against b1d7caac82c1cd6a9d142974f5e6dcbeff23a815 which has these src/runtime/contrib/cutlass/ sources (flash_decoding.cu, moe_gemm.cc), if these are duplicated now in the submodules and can be deleted then I can do that here. Though in my cursory read, I don't see that these are the same source files. Nonetheless, we can delete them if they are not needed. Let me know. https://github.com/apache/tvm/blob/b1d7caac82c1cd6a9d142974f5e6dcbeff23a815/cmake/modules/contrib/CUTLASS.cmake#L31-L33